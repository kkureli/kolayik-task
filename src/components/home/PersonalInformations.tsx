/* eslint-disable react-native/no-inline-styles */
import React, {FC} from 'react';
import {StyleSheet, View} from 'react-native';
import {User} from '../../models';
import Typography from '../common/Typography';

interface InfoProps {
  user: User;
}

interface RenderInfoProps {
  title: string;
  info: string;
  primary?: boolean;
}

export const RenderInformation: FC<RenderInfoProps> = ({
  title,
  info,
  primary
}) => {
  return (
    <View style={styles.infoRow}>
      <Typography gray>{title}</Typography>
      <Typography weight={'400'} primary={primary ? true : false}>
        {info}
      </Typography>
    </View>
  );
};

const PersonalInformations: FC<InfoProps> = ({
  user: {department, location, employmentStartDate, manager}
}) => {
  return (
    <View style={styles.container}>
      <RenderInformation title="Department" info={department} />
      <RenderInformation title="Location" info={location} />
      <RenderInformation
        title="Employment Start Date"
        info={employmentStartDate}
      />
      <RenderInformation primary title="Manager" info={manager} />
    </View>
  );
};

export default PersonalInformations;

const styles = StyleSheet.create({
  container: {
    marginTop: 24
  },
  infoRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginVertical: 10,
    marginHorizontal: 20
  }
});
