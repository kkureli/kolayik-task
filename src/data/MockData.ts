import {ContactMethods, SubMenuModel, User} from '../models';
const avatar = require('../assets/images/Avatar.png');

const mockUser: User = {
  image: avatar,
  name: 'Adnan Burhan',
  title: 'Product Designer',
  email: 'adnan@burhan.com',
  phoneNumber: 5360700195,
  department: 'Product',
  location: 'Istanbul',
  employmentStartDate: '25 Jan 2019',
  manager: 'Yunus Türkoğlu'
};

const contactOptions: ContactMethods[] = [
  {
    title: 'Email',
    icon: require('../assets/icons/contactOptions/emailIcon.png'),
    id: 'email'
  },
  {
    title: 'Call',
    icon: require('../assets/icons/contactOptions/phoneCallIcon.png'),
    id: 'call'
  }
];

const subMenu: SubMenuModel[] = [
  {
    title: 'Requests',
    icon: require('../assets/icons/subMenuIcons/requestsIcon.png'),
    id: '1',
    onPress: () => console.log('navigation.navigate(RequestsScreen)')
  },
  {
    title: 'Information',
    icon: require('../assets/icons/subMenuIcons/infoIcon.png'),
    id: '2',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Career',
    icon: require('../assets/icons/subMenuIcons/careerIcon.png'),
    id: '3',
    onPress: () => console.log('navigation.navigate(InformationScreen)')
  },
  {
    title: 'Leaves',
    icon: require('../assets/icons/subMenuIcons/leavesIcon.png'),
    id: '4',
    onPress: () => console.log('navigation.navigate(LeavesScreen)')
  },
  {
    title: 'Assets',
    icon: require('../assets/icons/subMenuIcons/assetsIcon.png'),
    id: '5',
    onPress: () => console.log('navigation.navigate(AssetsScreen)')
  },
  {
    title: 'Documents',
    icon: require('../assets/icons/subMenuIcons/documentsIcon.png'),
    id: '6',
    onPress: () => console.log('navigation.navigate(DocumentsScreen)')
  }
];

export {mockUser, subMenu, contactOptions};
