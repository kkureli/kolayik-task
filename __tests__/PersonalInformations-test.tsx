import React from 'react';
import renderer from 'react-test-renderer';
import PersonalInformations from '../src/components/home/PersonalInformations';
import {mockUser} from '../src/data/MockData';

test('PersonalInformationscComponent renders correctly', () => {
  const tree = renderer
    .create(<PersonalInformations user={mockUser} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
