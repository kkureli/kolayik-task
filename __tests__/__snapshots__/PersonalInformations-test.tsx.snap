// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInformationscComponent renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 24,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 10,
      }
    }
  >
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#535453",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      Department
    </Text>
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Product
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 10,
      }
    }
  >
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#535453",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      Location
    </Text>
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Istanbul
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 10,
      }
    }
  >
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#535453",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      Employment Start Date
    </Text>
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      25 Jan 2019
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 20,
        "marginVertical": 10,
      }
    }
  >
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#535453",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "normal",
        }
      }
    >
      Manager
    </Text>
    <Text
      margin={null}
      padding={null}
      style={
        Object {
          "color": "#376BFB",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Yunus Türkoğlu
    </Text>
  </View>
</View>
`;
